### PR TITLE
ci: add CI support via GitHub Actions

### PR DESCRIPTION
**Pull request by Alkaar**
Added CI support via GitHub Actions. This should allow running formatting, Scalafix, and tests for all PRs and pushes to master. Will need to keep this open for a bit to test it out.

- [x] Added CI via GitHub Actions config
- [x] Turned on more Scalafix rules
- [x] Fixed Scalafix bugs

### CODE REVIEW

File: project/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: CI
+
+on: [push, pull_request]
+#on:
+#  push:
+#    branches: [master]
+#  pull_request:
```
Comment (laughedelic) [inline project/.github/workflows/ci.yml:7]: I think this trigger syntax is wrong: `push` and `pull_request` are objects, so you should probably use the commented part
```suggestion
on:
  push:
    branches: [master]
  pull_request:
```

File: project/.github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: sbt update
+
+      - name: Compile all sources
+        run: sbt 'all compile Test/compile'
+
+      - name: Run Scalafix
+        run: sbt 'scalafixAll --check'
+
+      - name: Run tests
+        run: sbt 'all Test/test'
```
Comment (laughedelic) [inline project/.github/workflows/ci.yml:40]: Nit:
```suggestion
        run: sbt test
```

`all` is for running several tasks in parallel, here you run only one, and the scope is redundant

File: project/.github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Publish JUnit Test Report
+        if: always()
+        uses: mikepenz/action-junit-report@v3.3.3
+        with:
+          report_paths: "**/target/test-reports/*.xml"
+          require_tests: false
```
Comment (laughedelic) [inline project/.github/workflows/ci.yml:47]: Nit: if you have any tests, it's better to remove this option
```suggestion
```

File: project/.github/workflows/ci.yml (hunk 1 above)
Comment (Alkaar) [reply project/.github/workflows/ci.yml]: Figured out the issue. I had this file in the wrong directory which I fixed below 😅 
[project/.github/workflows/ci.yml → .github/workflows/ci.yml](https://github.com/Alkaar/resy-booking-bot/pull/38/commits/eed725d16c355310a49f009b3493486d0e2e3842#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f)

**Pull request merged**